arch/sim/src/sim/sim_checknuttxtypes.c

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+ * arch/sim/src/sim/sim_checknuttxtypes.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+
+/* nuttx types */
+
+#include <dirent.h>
+#include <sys/statfs.h>
+#include <sys/stat.h>
+
+/* hostfs types */
+
+#define __SIM__
+#include <nuttx/fs/hostfs.h>
+#undef __SIM__
+
+/* Here we make static assertions to ensure nuttx types (eg. struct stat)
+ * match the corresponding hostfs types. (eg. struct nuttx_stat_s)
```

Comment:
Is it dependent on host OS version, type (Linux, MacOS, FreeBSD, etc) or does it work same way for all OS?

Reply:
the latter.